### PR TITLE
Add support for multiple environment files

### PR DESCRIPTION
**Pull request by SteelPhase**
I've attempted to resolve the issue presented in #1746. I'm sure some things will need to match coding standards and etc, but this initial implementation appears to work with a single file, and an array of many files.

### CODE REVIEW

**Comment by SteelPhase** [comment]
Looks like gogetdoc isn't happy on osx 1.11.x test, but not sure it's related to my PR

File: package.json [hunk 1]
```diff
@@ -463,7 +463,13 @@
...
               "envFile": {
-                "type": "string",
+                "type": [
+                  "string",
+                  "array"
+                ],
+                "items": {
+                  "type": "string"
+                },
                 "description": "Absolute path to a file containing environment variable definitions.",
```
Comment (quoctruong) [inline package.json]: The description should be updated to reflect the fact that both string and array are supported now.

File: src/debugAdapter/goDebug.ts [hunk 2]
```diff
@@ -315,14 +315,23 @@ class Delve {
...
-			let fileEnv = {};
+			const envs = [process.env];
+
 			try {
-				fileEnv = parseEnvFile(launchArgs.envFile);
+				if (typeof launchArgs.envFile === 'string') {
+					envs.push(...parseEnvFiles([launchArgs.envFile]));
+				}
+
+				if (launchArgs.envFile instanceof Array) {
```
Comment (quoctruong) [inline src/debugAdapter/goDebug.ts]: You can also do `Array.isArray(launchArgs.envFile)`.

File: src/debugAdapter/goDebug.ts [hunk 3]
```diff
@@ -315,14 +315,23 @@ class Delve {
 			}
 
 			// read env from disk and merge into env variables
-			let fileEnv = {};
+			const envs = [process.env];
+
 			try {
-				fileEnv = parseEnvFile(launchArgs.envFile);
+				if (typeof launchArgs.envFile === 'string') {
+					envs.push(...parseEnvFiles([launchArgs.envFile]));
```
Comment (quoctruong) [inline src/debugAdapter/goDebug.ts]: I think we can also do `envs.push(parseEnvFile(launchArgs.envFile))`

File: src/goPath.ts [hunk 4]
```diff
@@ -142,6 +142,19 @@ export function parseEnvFile(path: string): { [key: string]: string } {
 	}
 }
 
+export function parseEnvFiles(paths: string[]): { [key: string]: string }[] {
+	const envs: { [key: string]: any }[] = [];
```
Comment (quoctruong) [inline src/goPath.ts]: I think you can shorten this function with `return paths.map(path => parseEnvFile(path));`

File: package.json (hunk 1 above)
Comment (SteelPhase) [reply package.json]: I've added a better description

**Comment by SteelPhase** [comment]
@quoctruong I've made the suggested changes.

**Comment by ramya-rao-a** [comment]
Apologies for not getting to this sooner @SteelPhase 
It somehow got dropped from my radar.

There have been a lot of changes to the `goDebug.ts` file.
Do you mind getting the latest from the master branch and resolving the conflicts here?

**Pull request merged**